metomi/rose/date.py

```diff
@@ -29,6 +29,9 @@
 from metomi.rose.reporter import Reporter
 
 
+LEGACY_OFFSET = re.compile(r'-?(?P<value>\d+)(?P<unit>[wdhms])')
```

Comment:
The `-` doesn't do anything with the way this is used
```suggestion
LEGACY_OFFSET = re.compile(r'(?P<value>\d+)(?P<unit>[wdhms])')
```

Reply:
Should the regex start ^ and end $ to enforce a full match?

Reply:
I don't think that would work with the way this is used (`.findall()`)

Reply:
then `re.match`?

Reply:
Would need to change the regex to something like `^-?((\d+)([wdhms]))+$`